checkov/terraform/checks/resource/aws/ECRPolicy.py

```diff
@@ -27,13 +27,25 @@ def scan_resource_conf(self, conf):
                 statement = policy['Statement'][0]
                 if statement['Principal'] and type(statement['Principal']) is str:
                     principal = statement['Principal']
-                    if principal == "*":
+                    if principal == "*"  and self.check_for_constrained_condition(statement) == False:
                         self.evaluated_keys = ["policy/Statement/Principal"]
                         return CheckResult.FAILED
         return CheckResult.PASSED
 
     def get_evaluated_keys(self) -> List[str]:
         return ['policy']
 
+    def check_for_constrained_condition(self, statement):
+        """
+        Checks to see if there is a constraint on a a wildcarded principal
+        :param statement: statement from aws_repository_configuration
+        :return: True if there is a constraint
+        """
+        if statement['Condition'] and type(statement['Condition']) is dict:
+          condition = statement['Condition']
+          if condition['ForAllValues:StringEquals'] and type(condition['ForAllValues:StringEquals']) is dict:
```

Comment:
same here, check if the key exists and then check if it is an instance of `dict`

Reply:
Done